src/api/pdc_client_connect.c

```
@@ -1629,8 +1631,11 @@ PDC_Client_finalize()
     if (HG_Context_destroy(send_context_g) != HG_SUCCESS)
         PGOTO_ERROR(FAIL, "Error with HG_Context_destroy");
 
-    if (HG_Finalize(send_class_g) != HG_SUCCESS)
-        PGOTO_ERROR(FAIL, "Error with HG_Finalize");
+    hg_ret = HG_Finalize(send_class_g);
+    if (pdc_client_mpi_rank_g == 0) {
+        if (hg_ret != HG_SUCCESS)
+            LOG_WARNING("Error with HG_Finalize\n");
+    }
```

We can propagate the error here. A previous PR on the develop branch free's the bulk handles so this no longer returns an error.

---

ok, will revert